api/envoy/extensions/filters/http/grpc_json_reverse_transcoder/v3/transcoder.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.grpc_json_reverse_transcoder.v3;
+
+import "google/protobuf/wrappers.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.grpc_json_reverse_transcoder.v3";
+option java_outer_classname = "TranscoderProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/grpc_json_reverse_transcoder/v3;grpc_json_reverse_transcoderv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: gRPC-JSON reverse transcoder]
+// gRPC-JSON reverse transcoder :ref:`configuration overview <config_http_filters_grpc_json_reverse_transcoder>`.
+// [#extension: envoy.filters.http.grpc_json_reverse_transcoder]
+
+// [#next-free-field: 6]
+// GrpcJsonReverseTranscoder is the filter configuration for the gRPC JSON
+// reverse transcoder. The reverse transcoder acts as a bridge between gRPC
+// client and an HTTP/JSON server, converting the gRPC request into HTTP/JSON
+// for the HTTP backend and the HTTP/JSON response back to gRPC for the gRPC
+// client.
+message GrpcJsonReverseTranscoder {
+  oneof descriptor_set {
+    option (validate.required) = true;
+
+    // Supplies the filename of
+    // :ref:`the proto descriptor set
+    // <config_grpc_json_generate_proto_descriptor_set>` for the gRPC services.
+    string descriptor_path = 1;
+
+    // Supplies the binary content of
+    // :ref:`the proto descriptor set
+    // <config_grpc_json_generate_proto_descriptor_set>` for the gRPC services.
+    bytes descriptor_binary = 2;
+  }
+
+  // The maximum size of a request body to be transcoded, in bytes.
+  //
+  // Large values may cause envoy to use a lot of memory if there are many
+  // concurrent requests.
+  //
+  // If unset, the current stream buffer size is used.
+  google.protobuf.UInt32Value max_request_body_size = 3 [(validate.rules).uint32 = {gt: 0}];
```

Comment:
under what situations would it make sense to not just change the max stream buffer size instead?

Reply:
I not sure if I understand what you're trying to ask, but `max_request_body_size` is the max buffer stream size. If this is not set the filter will use the default stream buffer limit.

Reply:
> max_request_body_size is the max buffer stream size

should this be renamed then to max_buffer_stream_size ?  what happens if the request body sent by the client is larger than this value?

Reply:
The filter buffers the request/response before forwarding it upstream/downstream. If the request/response exceeds the stream buffer limit then the filter stop the iteration and buffering and returns the local response with ResourceExhausted/Internal status. 

Here the max_request/response_body_size is the upper limit on the request/response sizes, which for filter means we don't wanna buffer the data more than the max allowed request/response sizes, that's why we're setting the buffer limits to the values of these fields.

Reply:
> The filter buffers the request/response before forwarding it upstream/downstream. If the request/response exceeds the stream buffer limit then the filter stop the iteration and buffering and returns the local response with ResourceExhausted/Internal status.

This is good info, thanks!  I suggest putting it in the comments for the field.  In that case, `max_request_body_size` makes sense for a name.

Reply:
Updated the comments on both the fields.